android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -418,4 +552,30 @@ private void clearAllSharedConfigsForView(View view) {
     mSnapshotRegistry.remove(viewTag);
     mNativeMethodsHolder.clearAnimationConfig(viewTag);
   }
+
+  private void cancelAnimation(View view) {
+    int viewTag = view.getId();
+    mNativeMethodsHolder.cancelAnimation(viewTag, "sharedTransition", true, true);
+  }
+
+  private void disableCleaningForViewTag(int viewTag) {
+    Integer counter = mDisableCleaningForViewTag.get(viewTag);
+    if (counter != null) {
+      mDisableCleaningForViewTag.put(viewTag, counter + 1);
+    } else {
+      mDisableCleaningForViewTag.put(viewTag, 1);
+    }
+  }
+
+  private void enableCleaningForViewTag(int viewTag) {
+    Integer counter = mDisableCleaningForViewTag.get(viewTag);
+    if (counter == null) {
+      return;
+    }
+    if (counter - 1 > 0) {
+      mDisableCleaningForViewTag.put(viewTag, counter - 1);
+    } else {
+      mDisableCleaningForViewTag.remove(viewTag);
+    }
```

Comment:
Analogously to iOS, but I'm okay with the current approach since it also works fine for negative numbers if we mess something up in the future.
```suggestion
    if (counter == 1) {
      mDisableCleaningForViewTag.remove(viewTag);
    } else {
      mDisableCleaningForViewTag.put(viewTag, counter - 1);
    }
```

Reply:
Resolved by: https://github.com/software-mansion/react-native-reanimated/pull/4043/commits/120c44ded503b3c38dc4219649190faf2cbc6214